WebApiCoreSeed.WebApi/Controllers/ValuesController.cs

```diff
@@ -1,52 +1,116 @@
-﻿namespace WebApiCoreSeed.WebApi.Controllers
-{
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Authorization;
+using WebApiCoreSeed.Data.EF;
+using WebApiCoreSeed.Data.Models;
+using WebApiCoreSeed.Domain.Services;
+using Microsoft.EntityFrameworkCore;
 
-    using System;
-    using System.Collections.Generic;
-    using System.Linq;
-    using System.Threading.Tasks;
-    using Microsoft.AspNetCore.Mvc;
-    using Microsoft.AspNetCore.Authorization;
 
+namespace WebApiCoreSeed.WebApi.Controllers
+{
 
+    
     [Route("api/[controller]")]
     public class ValuesController : Controller
     {
         // GET api/values
         [HttpGet]
-        public IEnumerable<string> Get()
+        public IActionResult Get()
         {
             //TODO: Add GetData from database
-            return new string[] { "value1", "value2" };
+            return Ok(new string[] { "value1", "value2" });
         }
 
-        // GET api/values/5
+        // GET api/values/0E95A953-D023-4425-ED35-08D4E34D4DB8
         [HttpGet("{id}")]
-        public string Get(string id)
+        public IActionResult Get(Guid id)
         {
-            //TODO: Add GetData from database
-            return "value";
+            User retrievedUser = null;
+            var myTask = GetAsync(id);
+            retrievedUser = myTask.Result;
+            if (retrievedUser == null)
+            {
+                return NotFound();
+            }
+                        
+            return Ok(retrievedUser);
+          
+        }
+
+
+
+        private async Task<User> GetAsync(Guid id)
+        {
+            var optionsBuilder = new DbContextOptionsBuilder<WebApiCoreSeedContext>();
+            User retrievedUser = null;                      

```

Comment:
Do you really need to declare this variable here? Perhaps you can just use `return await userService.GetByIdAsync(id)` inside the using. Or perhaps you were planning to add some error handling